File: base_page.py
from selenium.webdriver.support.ui import Select, WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class BasePage:

    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 40)

    def select_state(self, select_element, state):
        select = Select(select_element)
        self.wait.until(EC.element_to_be_clickable(select_element))
        select.select_by_visible_text(state)
        print(f"State '{state}' selected in dropdown")

File: src/main/java/com/example/BasePage.java
package com.example;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

public class BasePage {
    protected WebDriver driver;
    protected WebDriverWait wait;

    public BasePage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, 40);
    }

    public void selectState(WebElement selectElement, String state) {
        Select select = new Select(selectElement);
        wait.until(ExpectedConditions.elementToBeClickable(selectElement));
        select.selectByVisibleText(state);
        System.out.println("State '" + state + "' selected in dropdown");
    }
}